Release: 4.5.0-beta3

tridge opened the pull request:
Changes from 4.5.0-beta2

1) Board specific changes
- added PixFlamingo F7 board
- support ICM42688 on BlitzF745AIO

2) System level changes

- improved robustness of CRSF parser
- reduced memory used by DDS/ROS2
- added filesystem crc32 binding in lua scripting
- support visual odometry quality metric and added autoswitching lua script
- allow for expansion of fence storage to microSD for larger pologon fences
- allow FTP upload of fence and rally points
- fixed vehicle type of ship simulation for ship landing
- make severity level depend on generator error level in IE 2400 generator
- speed up initial GPS probe by using SERIALn_BAUD first
- allow NanoRadar radar and proximity sensor to share the CAN bus
- only produce *_with_bl.hex not *.hex in builds if bootloader available
- fixed check for GPS antenna separation in moving baseline yaw
- added GPS_DRV_OPTIONS options for fully parsing RTCMv3 stream
- fixed logging of RTCM fragments in GPS driver
- fixed video recording while armed
- robostness and logging improvements for ExternalAHRS

3) Plane specific changes
- added support for precision landing of quadplanes
- fixed speed tuning of fixed wing aircraft without an airspeed sensor
- improved pitch control in high speed VTOL flight
- added TECS option for rapid descent in fixed wing aircraft

4) Copter specific changes
- check fence breaches more often on copter for smaller overrun
- improved copter follow mode at close distances
- fixed default for FLTD for yaw
- fixed reset_target_and_rate method in attitude control

rmackay9 commented:
@tridge,

I've compared the commits included with the PRs they come from and found a few missing commits.  I suspect these are intentional but I'd like to be sure.

- AP_RCProtocol: correctly process bytes from standalone UART 	https://github.com/ArduPilot/ardupilot/pull/26261/commits
- AP_InertialSensor: fixed accel cal simple to remove unused IMUs	https://github.com/ArduPilot/ardupilot/pull/26329/commits
- AP_Filesystem: crc32: Ensure checksum value is initialized 	https://github.com/ArduPilot/ardupilot/pull/26211/commits
- AP_RCTelemetry: correct CSRF->CRSF				https://github.com/ArduPilot/ardupilot/pull/26258/commits
- autotest: fixed got and want ordering in text wait       https://github.com/ArduPilot/ardupilot/pull/26380/commits

rmackay9 commented:
Not critical but it looks like you've left the Copter specific changes in the Plane release notes. No harm of course but FYI

tridge commented:
@rmackay9 thanks!
I've corrected some of the errors, some were deliberate
 - the processing protocol bytes bug was folded into the general CRSF fixes, as is CRSF naming
 - the accel cal simple fix is now included
 - the crc32 init is now included
 - the autotest text order bug is not in 4.5
 - the rangefinder MSG fix is not user visible
 - i've added the MR72 addition to the notes
 - cuberedsecondary fix now in notes
 - the removing iomcu dshot is not user visible (we were using more flash space than needed)
 - i've added speedybee change to notes
